### PR TITLE
WX Medium Plan: Create 'Plans' Page

### PR DESCRIPTION
**Pull request by nelsonec87**
Related to https://github.com/Automattic/wp-calypso/issues/74126
~Depends on https://github.com/Automattic/wp-calypso/pull/74140~

## Proposed Changes

* Added logic to the man Plans component to identify sites using Woo Express plans
* Show a different page to Woo Express
* Hide the `My Plan | Plans` toggle for Woo Express
* Put these changes under the `plans/wooexpress-medium` feature flag

## Testing Instructions

* Get a site with the WX Medium plan
   * Start with a free site
   * Go to `/checkout/<site-slug>/wooexpress-medium-bundle-monthly`
   * Finish checkout
* Go to the Plans page (`/plans/<site-slug>`)
* With the feature flag enabled (`/plans/<site-slug>?flags=plans/wooexpress-medium`) you should see a WX placeholder
image

* With the feature flag disabled (`/plans/<site-slug>?flags=-plans/wooexpress-medium`) you should see the default Plans page.
image

### CODE REVIEW

File: client/my-sites/plans/main.jsx [hunk 1]
```diff
@@ -230,6 +234,29 @@ class Plans extends Component {
...
+			case PLAN_WOOEXPRESS_MEDIUM:
+			case PLAN_WOOEXPRESS_MEDIUM_MONTHLY:
+				if ( isEnabled( 'plans/wooexpress-medium' ) ) {
+					return this.renderWooExpressMediumPage();
+				}
+
+			default:
+				return this.renderPlansMain();
+		}
+	}
```
Comment (daledupreez) [inline client/my-sites/plans/main.jsx]: Might it be cleaner to handle this via early returns from multiple conditions? And maybe we can pass the `isWooExpressPlan` and `isECommerceTrial` flags from `render()` as arguments to simplify the logic as well?
```suggestion
	renderMainContent( { isECommerceTrial, isWooExpressPlan } ) {
		if ( isECommerceTrial ) {
			return this.renderEcommerceTrialPage();
		}

		if ( isWooExpressPlan && isEnabled( 'plans/wooexpress-medium' ) ) {
			return this.renderWooExpressMediumPage();
		}

		return this.renderPlansMain();
	}
```

**Review by daledupreez — COMMENTED** [review]
This is working as expected, but I think we can simplify the logic in `renderMainContent()`.

**Comment by nelsonec87** [comment]
Rebased to resolve conflicts and fix E2E tests.

**Review by daledupreez — APPROVED** [review]
Tested and LGTM!

Thanks for updating the logic! I think that's much simpler!

**Pull request merged**